facebook-core/src/main/java/com/facebook/internal/Utility.java

```diff
@@ -1186,4 +1187,25 @@ public static String generateRandomString(int length) {
         Random r = new Random();
         return new BigInteger(length * 5, r).toString(32);
     }
+
+    /*
+     * There is a bug on Android O that excludes the dialog's view hierarchy from the
+     * ViewStructure used by Autofill because the window token is lost when the dialog
+     * is resized, hence the token needs to be saved dialog is attached to a window and restored
+     * when the dialog attributes change after it is resized.
+     */
+    public static boolean mustFixWindowParamsForAutofill(Context context) {
+        // TODO: once this bug is fixed on Android P, checks for version here as well
+        return isAutofillAvailable(context);
+    }
+
+    public static boolean isAutofillAvailable(Context context) {
```

Comment:
Love that all this is in a class. Do we think the class should be a bit more specifically named? Maybe AutofillUtility or something to the like.

Reply:
Definitley just realized that I suggested renaming a class that we made eons ago. Doh! hah. This all looks pretty good. I'm still testing internally and will let you know how it shakes out.

Reply:
I actually just created that class but didn't update the pull request yet, so I will keep it on my branch.

Anyways, I will update the pull request with the fixed import (extra Utility in the other class)